wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -0,0 +1,1418 @@
+                    DYNAMIC_TYPE_TMP_BUFFER);
+        } 
+        else {
+            byte* pt = (byte*)XREALLOC(se050Ctx->msg, se050Ctx->used + len, se050Ctx->heap,
+                    DYNAMIC_TYPE_TMP_BUFFER);
+            if (pt == NULL) {
+                return -1;
+            }
+            se050Ctx->msg = pt;
+        }
+        se050Ctx->len = se050Ctx->used + len;
+    }
+    XMEMCPY(se050Ctx->msg + se050Ctx->used, data , len);
+    se050Ctx->used += len;
+    return 0;
+}
+
+int se050_hash_final(SE050_HASH_Context* se050Ctx, byte* hash, size_t digestLen, sss_algorithm_t algo)
+{
+    sss_status_t status;
+    sss_digest_t digest_ctx;
+    XMEMSET(&digest_ctx, 0, sizeof(digest_ctx));
+    
+    const byte*     data = se050Ctx->msg;
+    int             size = (se050Ctx->len) / SSS_BLOCK_SIZE;
+    int             leftover = (se050Ctx->len) % SSS_BLOCK_SIZE;
+    const byte*     blocks;
+                    blocks = data;    
+
+
+    if (wolfSSL_CryptHwMutexLock() == 0) { 
+        status = sss_digest_context_init(&digest_ctx, cfg_se050_i2c_pi, algo, kMode_SSS_Digest);  
+        if(status != kStatus_SSS_Success){
+            printf("error 1\n");
+            return -1;
+        }   
+
+        status = sss_digest_init(&digest_ctx);
+        if(status != kStatus_SSS_Success){
+            printf("error 2 - hash_final...\n");
+            return -1;
+        }
+        /* used to send chunks of size 512 */
+        while (size--) {
+            status = sss_digest_update(&digest_ctx, blocks, SSS_BLOCK_SIZE);
+            if(status != kStatus_SSS_Success){
+                printf("error 3\n");
+                return -1;
+            }
+            blocks += SSS_BLOCK_SIZE;       
+        }
+        if (leftover) {
+            status = sss_digest_update(&digest_ctx, blocks, leftover);
+            if(status != kStatus_SSS_Success){
+                printf("error 3\n");
+                return -1;
+            }
+        }        
+
+        status = sss_digest_finish(&digest_ctx, hash, &digestLen);
+        if(status != kStatus_SSS_Success){
+            printf("error 4\n");
+            return -1;
+        }
+        sss_digest_context_free(&digest_ctx);
+
+    }
+
+    wolfSSL_CryptHwMutexUnLock();
+
+    return 0;
+}
+
+void se050_hash_free(SE050_HASH_Context* se050Ctx)
+{
+    (void)se050Ctx;
+}
+
+#include <wolfssl/wolfcrypt/aes.h>
+int se050_set_key(Aes* aes, const byte* key, word32 len,
+                                        const byte* iv, int dir)
+{
+    //printf("ran se050_set_key\n");
+    if (aes == NULL ||  !( (len == 16) || (len == 24) || (len == 32)) ) {
+            return BAD_FUNC_ARG;
+    }
+
+    (void)dir;
+    (void)iv;
+    sss_status_t status;
+    aes->rounds = len/4 + 6;
+    sss_object_t newKey;
+    sss_key_store_t host_keystore = {0};
+    uint32_t keyId = rand(); /*58;*/ //maybe set this equal to a rng so that it can be used multiple times without erroring out...
```

Comment:
You cannot use `rand()`. That is a stdlib implementation. Either cleanup to use the key allocator logic or change to use our `wc_RNG_GenerateBlock`.